pkgs/applications/misc/1password/default.nix

```diff
@@ -1,50 +1,57 @@
-{ stdenv, fetchzip, autoPatchelfHook, fetchurl, xar, cpio }:
+{ stdenv, fetchzip, fetchurl, autoPatchelfHook, xar, cpio, installShellFiles }:
 
 stdenv.mkDerivation rec {
   pname = "1password";
-  version = "1.1.1";
+  version = "1.2.0";
   src =
     if stdenv.isLinux then fetchzip {
       url = {
         "i686-linux" = "https://cache.agilebits.com/dist/1P/op/pkg/v${version}/op_linux_386_v${version}.zip";
         "x86_64-linux" = "https://cache.agilebits.com/dist/1P/op/pkg/v${version}/op_linux_amd64_v${version}.zip";
       }.${stdenv.hostPlatform.system};
       sha256 = {
-        "i686-linux" = "1andl3ripkcg4jhwdkd4b39c9aaxqpx9wzq21pysn6rlyy4hfcb0";
-        "x86_64-linux" = "0qj5v8psqyp0sra0pvzkwjpm28kx3bgg36y37wklb6zl2ngpxm5g";
+        "i686-linux" = "1fpmglqj7qqaa7k79ngrf16qq6rv4m8b57hzq87fyj22myaxydjy";
+        "x86_64-linux" = "0c0zj3bv0rgdqaanm6gk7fsy5ckj5h8zc5jckrv1nsfqr7lpbydd";
       }.${stdenv.hostPlatform.system};
       stripRoot = false;
     } else fetchurl {
       url = "https://cache.agilebits.com/dist/1P/op/pkg/v${version}/op_darwin_amd64_v${version}.pkg";
-      sha256 = "16inwxkrky4xwlr7vara1l8kapdgjg3kfq1l94i5855782hn4ppm";
+      sha256 = "0sc5fi55gcbjmj5cp9clilkdz7c6nzqyyh1fficfafh8il4qvki6";
     };
 
+  nativeBuildInputs = [ installShellFiles ]
+  ++ stdenv.lib.optionals stdenv.isLinux [ autoPatchelfHook ];
+
   buildInputs = stdenv.lib.optionals stdenv.isDarwin [ xar cpio ];
 
   unpackPhase = stdenv.lib.optionalString stdenv.isDarwin ''
     xar -xf $src
     zcat Payload | cpio -i
   '';
 
+  dontStrip = stdenv.isDarwin;
+
   installPhase = ''
     install -D op $out/bin/op
   '';
 
-  dontStrip = stdenv.isDarwin;
-
-  nativeBuildInputs = stdenv.lib.optionals stdenv.isLinux [ autoPatchelfHook ];
-
   doInstallCheck = true;
 
   installCheckPhase = ''
     $out/bin/op --version
   '';
 
+  postInstall = ''
+    $out/bin/op completion bash > op.bash
+    $out/bin/op completion zsh > op.zsh
+    installShellCompletion op.{bash,zsh}
+  '';
```

Comment:
I didn't look at the rest of the derivation yet and it seems that it already has an `installPhase`, so these can just go there.

Reply:
Hmpf, that's annoying, this is a precompiled binary, so it's not actually runnable yet. Moreover, moving this to `postFixup` doesn't work either, because auto-patchelf also runs during `postFixup`.

There are three obvious options, but I don't have any strong preferences:

1. Do not ship the shell completions.
2. Run autoPatchelf in another phase.
3. Use a `patchPhase` that does everything in the correct order.

Anyone?